lang/en/auth_saml2.php

```diff
@@ -33,6 +33,12 @@
 $string['auth_saml2blockredirectdescription'] = 'Redirect or display message to SAML2 logins based on configured group restrictions';
 $string['autocreate'] = 'Auto create users';
 $string['autocreate_help'] = 'If users are in the IdP but not in moodle create a moodle account.';
+$string['autologin'] = 'Auto-login';
+$string['autologin_help'] = 'On pages that allow guest access without login, automatically log users into Moodle with a real user account if they are logged in to the IdP (using passive authentication).';
```

Comment:
Hi @sammarshallou ! Thank you for you changes. 
Sorry but I maybe miss something, but I'm still not 100% clear why we apply autologin on pages that do not require login at all and ignore others? I would think that this would be helpful for IDPs (e.g. like OKTA) that do not support autologin.

Reply:
Yes, this feature is intended to only run on pages that don't require login (either because they literally don't call require_login, or because they allow automatic guest login).

The reason for this is that if the page isn't one of these, i.e. if it DOES require login (& not allow guest autologin), then the user will be redirected to the login screen. And auth_saml2 already has an automatic login feature once you get to the login screen (set 'duallogin' to 'Passive mode' if you also allow other logins, or 'No' if you only allow login from the one SAML IdP). So I didn't want to overlap with the existing feature which already works well.

In terms of whether it works on different IdPs (I don't know about OKTA sorry), this new feature requires passive mode support from the IdP, the same as the existing automatic login from login screen, so I don't think there should be any difference in terms of support.